cloud/azure/custom/api.pm

```diff
@@ -790,6 +790,56 @@ sub azure_get_publicip {
     return $response;
 }
 
+sub azure_get_budget_set_url {
+    my ($self, %options) = @_;
+
+    my $url = $self->{management_endpoint} . "/subscriptions/" . $self->{subscription};
+    $url .= "/resourceGroups/" . $options{resource_group} if (defined($options{resource_group}) && $options{resource_group} ne '');
+    $url .= "/providers/Microsoft.Consumption/budgets/" . $options{budget_name} if (defined($options{budget_name}) && $options{budget_name} ne '');
```

Comment:
Budget name is already mandatory at the mode level, I think we don't need another if defined(budget) check right?